pkg/json2table/json2table.go

```diff
@@ -0,0 +1,89 @@
+package json2table
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Jeffail/gabs/v2"
+)
+
+func ConvertJSONToTable(rootPath string, jsonPaths map[string]string, rootObj interface{}) ([]map[string]interface{}, error) {
```

Comment:
Can you add some comments, detailing what each variable should be? Also it would be nice to have more descriptive argument names. For example `rootObj` -> `jsonOrStructObject`/`sourceData`, `rootPath` -> `jsonPathsStartingPoint`/`jsonPathsPrefix`/`rootLocator`

Reply:
```suggestion
// ConvertJSONToTable extracts data from a JSON or struct object using specified paths.
//
// jsonPathsPrefix: Points to a specific location in the JSON or struct object from where 
// extraction should begin. If left empty, extraction will start from the root of the object.
//
// columnPathMapping: A map where each key represents the desired column name in the output 
// table, and each value represents a gabs path to extract the data within the given JSON or struct.
//
// sourceData: JSON or struct from which data extraction should take place.
//
// Returns a slice of maps. Each map represents a row in the table, with each key-value 
// pair in the map being equivalent to a column name and its value.
```